parser/src/java/org/apache/hadoop/hive/ql/parse/ReplClauseParser.g

```diff
@@ -0,0 +1,110 @@
+/**
+   Licensed to the Apache Software Foundation (ASF) under one or more
+   contributor license agreements.  See the NOTICE file distributed with
+   this work for additional information regarding copyright ownership.
+   The ASF licenses this file to You under the Apache License, Version 2.0
+   (the "License"); you may not use this file except in compliance with
+   the License.  You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+parser grammar ReplClauseParser;
```

Comment:
Will we consider using Antlr4 in the future?  IMO, most of OSS use Antlr4, and only Hive uses Antlr3, and Antlr has not been updated for many years & Antlr4 is more easy to use than Antlr3.
It would be great if we could incrementally update `HiveParser.g` it to a higher version `HiveParser.g4`. :)

This comment may go beyond the scope of the PR. We can make this change happen in the future.

Reply:
Yahh, there are some efforts going on for Antlr4, but that might take some time. But we should definitely give it a shot, will explore how things can move there

this change anyway shouldn’t impact that upgrade efforts I believe